types/instance.d.ts

```diff
@@ -0,0 +1,119 @@
+import * as http from 'http'
+import * as http2 from 'http2'
+import * as https from 'https'
+
+import { InjectOptions, InjectPayload } from 'light-my-request'
+import { RouteOptions, RouteShorthandMethod } from './route'
+import { FastifySchema, FastifySchemaCompiler } from './schema'
+import { RawServerBase, RawRequestBase, RawReplyBase, RawServerDefault, RawRequestDefault, RawReplyDefault, ContextConfigDefault, RequestBodyDefault, RequestQuerystringDefault, RequestParamsDefault, RequestHeadersDefault } from './utils'
+import { FastifyLoggerOptions } from './logger'
+import { FastifyRegister } from './register'
+import { onCloseHook, onRouteHook, onRequestHook, onSendHook, onErrorHook, preHandlerHook, preParsingHook, preSerializationHook, preValidationHook } from './hooks'
+import { FastifyRequest } from './request'
+import { FastifyReply } from './reply'
+import { FastifyError } from './error'
+import { addContentTypeParser, hasContentTypeParser } from './content-type-parser';
+
+/**
+ * Fastify server instance. Returned by the core `fastify()` method.
+ */
+export interface FastifyInstance<
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestBase = RawRequestDefault<RawServer>,
+  RawReply extends RawReplyBase = RawReplyDefault<RawServer>,
+  Logger = FastifyLoggerOptions<RawServer>
+> {
+  server: RawServer
+  prefix: string
+  log: Logger
+
+  addSchema(schema: FastifySchema): FastifyInstance<RawServer, RawRequest, RawReply>
+
+  after(err: Error): FastifyInstance<RawServer, RawRequest, RawReply>
+
+  close(closeListener?: () => void): void
+  close<T>(): Promise<T> // what is this use case? Not documented on Server#close
+
+  // should be able to define something useful with the decorator getter/setter pattern using Generics to enfore the users function returns what they expect it to
+  decorate(property: string, value: any, dependencies?: Array<string>): FastifyInstance<RawServer, RawRequest, RawReply>
+  decorateRequest(property: string, value: any, dependencies?: Array<string>): FastifyInstance<RawServer, RawRequest, RawReply>
+  decorateReply(property: string, value: any, dependencies?: Array<string>): FastifyInstance<RawServer, RawRequest, RawReply>
```

Comment:
Add `symbol` type for the property parameter on all three decorate methods. Additionally, investigate if its possible to extend the FastifyInstance type dynamically through these methods